nominal/core.py

```diff
@@ -311,6 +339,132 @@ def _from_conjure(cls, client: NominalClient, dataset: scout_catalog.EnrichedDat
         )
 
 
+@dataclass(frozen=True)
+class LogSet:
+    rid: str
+    created_by: str
+    name: str
+    created_at: datetime
+    updated_at: datetime
+    log_count: int
+    timestamp_type: TimestampType
+    origin_metadata: Mapping[str, str]
+    description: str | None
+    _client: NominalClient = field(repr=False)
+
+    def add_logs_and_finalize(self, logs: list[Log]) -> Self:
```

Comment:
there's maybe an opportunity to have a builder pattern here where the user can keep adding logs before we send out a network request and finalize the log set, but I'm not sure whether it's worth the effort given that it might get ripped out. Key API weakness here is log sets are immutable once you attach a set of logs once

Reply:
allow for more flexible / less restrictive types on inputs. We don't need a `list` specifically, just something we can iterate through (i.e. an iterable)

Reply:
also agree that a builder may make sense for this, for the same reasons it makes sense for checklists

Reply:
resolved during pairing, consolidating into one method with create.